test_provider.py

```diff
@@ -0,0 +1,310 @@
+import json
+from datetime import datetime, timedelta
+from common.api import ProgramProvider
+from common.minimodel import *
+
+
+class JsonProvider(ProgramProvider):
+
+    # TODO: To handle Enum it could be done this way or changing the Enum to accept the 
+    # string value and just use the constructor. Let me know which is better.
+
+    obs_classes = {'partnerCal': ObservationClass.PARTNER_CAL,
+                   'science': ObservationClass.SCIENCE,
+                   'programCal': ObservationClass.PROG_CAL,
+                   'acq': ObservationClass.ACQ,
+                   'acqCal': ObservationClass.ACQ_CAL,
+                   'dayCal': None}
+    qa_states = {'Pass': QAState.PASS,
+                'Fail': QAState.FAIL,
+                'Usable': QAState.USABLE,
+                'Undefined': QAState.UNDEFINED}
+
+    def __init__(self, path):
+        self.path = path
+
+    def load_program(self, path: str) -> dict:
+        with open(path, 'r') as f:
+            return json.loads(f.read())
+    
+    @staticmethod
+    def sort_observations(obs: List[Observation]) -> List[Observation]:
+        return sorted(obs, key=lambda x: x.order)
+
+    @staticmethod
+    def parse_magnitude(json: dict) -> Magnitude:
+        band = MagnitudeBand(json['name'], None, None)
```

Comment:
For `MagnitudeBand`, we should never have to create these. We should just be selecting the one with the given `name` from `MagnitudeBands`:
```
band =MagnitudeBands[json['name']]
```
Then create the `Magnitude` like this:
```
mag = Magnitude(band, json['value'], None)
```
We probably do want to assert that the `system` matches the `system` in the `MagnitudeBand` that we retrieve and if it doesn't, log and raise a `ValueError`.

Reply:
I added comments to the mini-model to make this more explicit, and will add some more comments because perhaps my design choices might be a bit unclear.